app/scripts/lib/xhr.js

```diff
@@ -92,6 +93,34 @@ define([
      */
     getJSON: function (url, data, success) {
       return p.jQueryXHR($.getJSON(url, data, success));
+    },
+
+    oauthAjax: function (options) {
+      var request = {
+        url: options.url,
+        // make sure to set the dataType for Firefox <21. See issue #1930
+        dataType: 'json',
+        type: options.type,
+        headers: {
+          Authorization: 'Bearer ' + options.accessToken,
+          Accept: 'application/json'
+        }
+      };
+
+      if (options.headers) {
+        _.extend(request.headers, options.headers);
+      }
+
+      var data = options.data;
+      if (data) {
+        request.data = data;
+      }
+
+      if (typeof Blob !== 'undefined' && data instanceof Blob) {
+        request.processData = false;
```

Comment:
This line was not tested when it was in profile-client, so I'm not going to worry too much about it.